backend/utils.py

```
@@ -158,15 +168,19 @@ def process_uploaded_image_sync(file: UploadFile) -> tuple[Image.Image, bytes]:
-                detail=f"Invalid file type. Only image files are allowed. Detected: {detected_mime}"
-            )
+        if HAS_MAGIC:
+            file_content = file.file.read(1024)
+            file.file.seek(0)
+            try:
+                detected_mime = magic.from_buffer(file_content, mime=True)
+                if detected_mime not in ALLOWED_MIME_TYPES:
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Invalid file type. Only image files are allowed. Detected: {detected_mime}"
+                    )
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)